### PR TITLE
AMP-46510 Java builders

### PR DESCRIPTION
**Pull request by falconandy**

### CODE REVIEW

**Review by justin-fiedler — CHANGES_REQUESTED** [review]
@falconandy looks good but I spoke with the team and there was a heavy preference to keep the `MyEvent.builder()` method in Java. Main reason being it is a common, expected pattern in Java albeit more verbose.

Another benefits of `builder()` method:
- Single entry point for all Event classes (always SomeEvent.builder(), vs SomeEvent.someProperty())
- Keep the generated Event classes more compact for Events with only optional params
- Builders can be overridden in a single location in the case of subclassing

**Comment by falconandy** [comment]
@justin-fiedler `.builder()` is restored

File: android/java/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.java [hunk 1]
```diff
@@ -133,20 +133,20 @@ public void track(Event event, EventOptions options, MiddlewareExtra extra) {
         this.client.logEvent(event.eventType, this.getEventPropertiesJson(event), extra);
     }
 
-    public void identify(String userId, IdentifyProperties properties) {
+    public void identify(String userId, Identify properties) {
```
[the remark this answers is not in the document]
Comment (justin-fiedler) [reply android/java/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.java:136]: Ideally in the future we will have Builders for both Properties and Events. As discussed before to support strong object properties well we will want something like below. Given Java's verbosity I like the builders you implemented on the Event classes (compared to making new Event Property classes with own builders.) In the future we can create Property classes + builders for Property Groups.

```
MyEvent.builder().requiredProp(1).propertyGroupProp(
  PropertyGroupProp.builder().foo(true).build()
).optProp("cool").build()
```

**Review by justin-fiedler — CHANGES_REQUESTED** [review]
Overall looks good, see comments.

File: android/java/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (falconandy) [reply android/java/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.java]: Renamed to `event`.
We will need builders for nested typed properties. We can still use event builders for top level.

**Review by justin-fiedler — APPROVED** [review]

**Pull request merged**